crates/uv/src/commands/python/pin.rs

```diff
@@ -322,22 +323,26 @@ struct Pin<'a> {
 
 /// Checks if the pinned Python version is compatible with the workspace/project's `Requires-Python`.
 fn assert_pin_compatible_with_project(pin: &Pin, virtual_project: &VirtualProject) -> Result<()> {
+    // Don't factor in requires-python settings on dependency-groups
```

Comment:
Why not? What happens if you do `uv python pin 3.12` but a dependency group requires `>=3.13`? What if it's the `dev` group?

Reply:
I believe my concern here was if we do this we'll have to add all the dependency-group flags to `pin`

Reply:
I don't think you _need_ to add the group flags, per-say. It's just used to power a warning, right? You probably don't want your pinned Python version to be incompatible with your default groups. I'm not even sure you want it to be incompatible with your non-default groups? because then `--group foo` will fail without an explicit `-p <version>` which is super annoying.

Reply:
So I don't think we should ever use the intersection of --all-groups without the user asking, because although it's very dubious for this to come up, I think it should be allowed for two groups to have conflicting requires-pythons.

There is one place in the code where assert_pin_compatible_project is `?`ed instead of immediately discarded as a warning:

```
if let Some(virtual_project) = &virtual_project {
        if let Some(request_version) = pep440_version_from_request(&request) {
             assert_pin_compatible_with_project(
```